### PR TITLE
[somfytahoma] code cleanup

### PR DESCRIPTION
**Pull request by octa22**
This PR brings code simplification and clean up for the Somfy Tahoma binding

### CODE REVIEW

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java [hunk 1]
```diff
@@ -234,71 +231,27 @@ private void setTooManyRequests() {
...
-            logger.debug("Cannot get Tahoma events!", e);
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
-            if (e instanceof InterruptedException) {
-                Thread.currentThread().interrupt();
-            }
-        }
-        return new ArrayList<>();
+        SomfyTahomaEvent[] response = invokeCallToURL(TAHOMA_API_URL + "events/" + eventsId + "/fetch", "",
+                HttpMethod.POST, SomfyTahomaEvent[].class);
+        return response != null ? new ArrayList<>(Arrays.asList(response)) : new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java:254]: Fixed.

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java [hunk 2]
```diff
@@ -913,28 +676,48 @@ public void handleConfigurationUpdate(Map<String, Object> configurationParameter
     }
 
     public synchronized void refresh(String url, String stateName) {
+        SomfyTahomaState state = invokeCallToURL(DEVICES_URL + urlEncode(url) + "/states/" + stateName, "",
+                HttpMethod.GET, SomfyTahomaState.class);
+        if (state != null && StringUtils.isNotEmpty(state.getName())) {
```
[the remark this answers is not in the document]
Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java:681]: ok, fixed

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java [hunk 3]
```diff
@@ -913,28 +676,48 @@ public void handleConfigurationUpdate(Map<String, Object> configurationParameter
...
+            updateDevice(url, Arrays.asList(state));
```
[the remark this answers is not in the document]
Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java:682]: Fixed

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java [hunk 4]
```diff
@@ -234,71 +231,27 @@ private void setTooManyRequests() {
...
-            }
-        } catch (InterruptedException | TimeoutException e) {
-            logger.info("Cannot register events!", e);
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
-            if (e instanceof InterruptedException) {
-                Thread.currentThread().interrupt();
-            }
+            return URLEncoder.encode(text, StandardCharsets.UTF_8.toString());
+        } catch (UnsupportedEncodingException e) {
+            return text;
```
[the remark this answers is not in the document]
Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java:243]: yes, this is intended. From my point of view there is no need to put thing OFFLINE, rather return unencoded string

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java [hunk 5]
```diff
@@ -356,103 +309,19 @@ private void stopPolling() {
...
-            logger.debug("Cannot send get devices command!", e);
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
-            if (e instanceof InterruptedException) {
-                Thread.currentThread().interrupt();
-            }
-        }
-        return Collections.emptyList();
+        SomfyTahomaDevice[] response = invokeCallToURL(SETUP_URL + "devices", "", HttpMethod.GET,
+                SomfyTahomaDevice[].class);
+        return response != null ? Arrays.asList(response) : Collections.emptyList();
```
Comment (J-N-K) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java:324]:
```suggestion
        return response != null ? List.of(response) : List.of();
```

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java [hunk 6]
```diff
@@ -234,71 +231,27 @@ private void setTooManyRequests() {
...
-            logger.debug("Cannot get Tahoma events!", e);
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
-            if (e instanceof InterruptedException) {
-                Thread.currentThread().interrupt();
-            }
-        }
-        return new ArrayList<>();
+        SomfyTahomaEvent[] response = invokeCallToURL(TAHOMA_API_URL + "events/" + eventsId + "/fetch", "",
+                HttpMethod.POST, SomfyTahomaEvent[].class);
+        return response != null ? new ArrayList<>(List.of(response)) : List.of();
```
Comment (J-N-K) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java:254]:
```suggestion
        return response != null ? List.of(response) : List.of();
```

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java [hunk 7]
```diff
@@ -356,103 +309,19 @@ private void stopPolling() {
...
-            logger.debug("Cannot send getActionGroups command!", e);
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
-            if (e instanceof InterruptedException) {
-                Thread.currentThread().interrupt();
-            }
-        }
-        return null;
+        SomfyTahomaActionGroup[] list = invokeCallToURL(TAHOMA_API_URL + "actionGroups", "", HttpMethod.GET,
+                SomfyTahomaActionGroup[].class);
+        return list != null ? Arrays.asList(list) : new ArrayList<>();
```
Comment (J-N-K) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java:314]:
```suggestion
        return list != null ? List.of(list) : List.of();
```

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java [hunk 8]
```diff
@@ -693,54 +562,29 @@ public void sendCommand(String io, String command, String params) {
...
-                registerExecution(io, data.getExecId());
+    private Boolean sendCommandInternal(String io, String command, String params) {
+        String value = params.equals("[]") ? command : params.replace("\"", "");
+        String urlParameters = "{\"label\":\"" + getThingLabelByURL(io) + " - " + value
+                + " - OH2\",\"actions\":[{\"deviceURL\":\"" + io + "\",\"commands\":[{\"name\":\"" + command
+                + "\",\"parameters\":" + params + "}]}]}";
+        SomfyTahomaApplyResponse response = invokeCallToURL(EXEC_URL + "apply", urlParameters, HttpMethod.POST,
+                SomfyTahomaApplyResponse.class);
+        if (response != null) {
+            if (!StringUtils.isEmpty(response.getExecId())) {
```
Comment (J-N-K) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java:578]:
```suggestion
            if (!response.getExecId().isEmpty()) {
```

**Review by J-N-K — CHANGES_REQUESTED** [review]

**Comment by J-N-K** [comment]
I think you can just press "commit suggestion" if you agree. Don't care about the failing DCO, I'll override that.

**Comment by octa22** [comment]
> I think you can just press "commit suggestion" if you agree. Don't care about the failing DCO, I'll override that.

OK, I did it as you'd suggested. Thanks.

**Comment by J-N-K** [comment]
DCO overrided- all commit properly signed.off

**Review by J-N-K — APPROVED** [review]

**Comment by J-N-K** [comment]
somfytahome build succeeded in jenkins

**Pull request merged**